test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -256,7 +257,7 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 				Expect(err).NotTo(HaveOccurred())
 				Expect(pass).To(BeTrue())
 				By("Validating the omsagent daemonset")
-				running, err = pod.WaitOnReady("omsagent", "kube-system", 3, 1*time.Second, cfg.Timeout)
+				running, err = pod.WaitOnReady("omsagent", "kube-system", kubeSystemPodsReadinessChecks, 1*time.Second, cfg.Timeout)
```

Comment:
all pod checks seem to be hardcoded to 3 so can we change kubeSystemPodsReadinessChecks to defaultPodsReadinessChecks and use it in all occurrences or make a separate const for readiness checks for pods in the default namespace?

Reply:
I'm only touching `kube-system` pods intentionally, as those pods have a higher likelihood of being intertwined at boot time (i.e., until a quorum of foundational componentry is up, certain other pods may not be able to come online). In other words, it's "normal" for kube-system pods to have a few restarts during initial cluster creation, and so we should be more tolerant of that normal behavior. We observe this in test flakes.

The reason not to increase everything to a higher number is that, arguably, it would slow down the test unnecessarily (as you increase the number of required readiness checks you increase the time it takes to validate that particular pod). Also, for "on-demand" workloads (e.g., nginx), we arguably want to be less tolerant if a pod is thrashing.